nipype/pipeline/engine/tests/test_newnode_neuro.py

```diff
@@ -0,0 +1,120 @@
+from .. import NewNode, NewWorkflow
+from ..auxiliary import Function_Interface
+
+#dj niworkflows vs ...??
+from niworkflows.nipype.interfaces.utility import Rename
+import niworkflows.nipype.interfaces.freesurfer as fs
+
+from ...interfaces.freesurfer import PatchedConcatenateLTA as ConcatenateLTA
+
+Name = "example"
+DEFAULT_MEMORY_MIN_GB = None
+Inputs = {}
+
+# wf = Workflow(name, mem_gb_node=DEFAULT_MEMORY_MIN_GB,
+#               inputs=['source_file', 't1_preproc', 'subject_id',
+#                       'subjects_dir', 't1_2_fsnative_forward_transform',
+#                       'mem_gb', 'output_spaces', 'medial_surface_nan'],
+#               outputs='surfaces')
+#
+#dj: why do I need outputs?
+wf = NewWorkflow(name=Name, mem_gb_node=DEFAULT_MEMORY_MIN_GB, inputs=Inputs)
+
+
+# @interface
+# def select_target(subject_id, space):
+#     """ Given a source subject ID and a target space, get the target subject ID """
+#     return subject_id if space == 'fsnative' else space
+
+def select_target(subject_id, space):
+    """ Given a source subject ID and a target space, get the target subject ID """
+    return subject_id if space == 'fsnative' else space
+
+#select_target_interface = Function_Interface(select_target, ["out"])
+
+
+# wf.add('targets', select_target(subject_id=wf.inputs.subject_id))
+#   .map('space', space=[space for space in wf.inputs.output_spaces
+#                        if space.startswith('fs')])
+
+#dj: don't have option in map to connect with wf input
+
+wf.add(runnable=select_target, name="targets", subject_id="subject_id")\
+    .map(mapper="space", inputs={"space": [space for space in Inputs["output_spaces"]
+                                           if space.startswith("fs")]})
```

Comment:
this should become `wf.inputs.output_spaces`, so that such functionality can be implemented. this is also a place where we need to think about how these future inputs should be evaluated. since we don't know the length in advance, the list expansion is not going to work properly.

Reply:
will add option to call `connect` in `add`